hedera-node/src/main/java/com/hedera/services/queries/token/GetTokenNftInfosAnswer.java

```diff
@@ -0,0 +1,187 @@
+import com.hederahashgraph.api.proto.java.Query;
+import com.hederahashgraph.api.proto.java.Response;
+import com.hederahashgraph.api.proto.java.ResponseCodeEnum;
+import com.hederahashgraph.api.proto.java.TokenGetNftInfosQuery;
+import com.hederahashgraph.api.proto.java.TokenGetNftInfosResponse;
+import com.hederahashgraph.api.proto.java.TokenNftInfo;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static com.hedera.services.utils.SignedTxnAccessor.uncheckedFrom;
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.INVALID_NFT_ID;
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.INVALID_QUERY_RANGE;
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.INVALID_TOKEN_ID;
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.NOT_SUPPORTED;
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.OK;
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.TOKEN_WAS_DELETED;
+import static com.hederahashgraph.api.proto.java.ResponseType.COST_ANSWER;
+
+public class GetTokenNftInfosAnswer implements AnswerService {
+
+	public static final String TOKEN_NFT_INFOS_CTX_KEY = GetTokenNftInfosAnswer.class.getSimpleName() + "_tokenNftInfos";
+
+	private OptionValidator validator;
+
+	public GetTokenNftInfosAnswer(OptionValidator validator) {
+		this.validator = validator;
+	}
+
+	@Override
+	public boolean needsAnswerOnlyCost(Query query) {
+		return COST_ANSWER == query.getTokenGetNftInfos().getHeader().getResponseType();
+	}
+
+	@Override
+	public boolean requiresNodePayment(Query query) {
+		return typicallyRequiresNodePayment(query.getTokenGetNftInfos().getHeader().getResponseType());
+	}
+
+	@Override
+	public Response responseGiven(Query query, StateView view, ResponseCodeEnum validity, long cost) {
+		return responseFor(query, view, validity, cost, NO_QUERY_CTX);
+	}
+
+	@Override
+	public Response responseGiven(
+			Query query, StateView view, ResponseCodeEnum validity, long cost, Map<String, Object> queryCtx) {
+		return responseFor(query, view, validity, cost, Optional.of(queryCtx));
+	}
+
+	@Override
+	public ResponseCodeEnum checkValidity(Query query, StateView view) {
+		var tokenNftInfosQuery = query.getTokenGetNftInfos();
+		var id = tokenNftInfosQuery.getTokenID();
+
+		var optionalInfo = view.infoForToken(id);
+		if (!optionalInfo.isPresent()) {
+			return INVALID_TOKEN_ID;
+		}
+
+		if (optionalInfo.get().getDeleted()) {
+			return TOKEN_WAS_DELETED;
+		}
+
+		var optionalId = view.tokenWith(id);
+		if (optionalId.isPresent() && !optionalId.get().tokenType().equals(TokenType.NON_FUNGIBLE_UNIQUE)) {
+			return NOT_SUPPORTED;
+		}
+
+		if (tokenNftInfosQuery.getStart() >= tokenNftInfosQuery.getEnd()) {
+			return INVALID_QUERY_RANGE;
+		}
+
+		var validity = validator.nftMaxQueryRangeCheck(
+				tokenNftInfosQuery.getStart(), tokenNftInfosQuery.getEnd());
+		if (validity != OK) {
+			return validity;
+		}
```

Comment:
Failing fast if there is an error is a priority, that being said I think that it is better to place those 2 checks before we load the Token from the state